readthedocs_ext/readthedocs.py

```diff
@@ -126,6 +149,45 @@ def rtd_render(self, template, render_context):
                                                         app.builder.templates)
 
 
+def generate_json_artifacts(app, pagename, templatename, context, doctree):
+    """
+    Generate JSON artifacts for each page.
+
+    This way we can skip generating this in other build step.
+    """
+    try:
+        if not app.config.rtd_generate_json_artifacts:
+            return
+        # We need to get the output directory where the docs are built
+        # _build/json.
+        build_json = os.path.abspath(
+            os.path.join(app.outdir, '..', 'json')
+        )
+        outjson = os.path.join(build_json, pagename + '.fjson')
+        outdir = os.path.dirname(outjson)
+        if not os.path.exists(outdir):
+            os.makedirs(outdir)
+        with open(outjson, 'w+') as json_file:
+            to_context = {
+                key: context[key]
+                for key in KEYS
+                if key in context
+            }
+            json_file.write(json.dumps(to_context, indent=4))
```

Comment:
what about using `json.dump` instead of `dumps`?
`json.dump(to_context, json_file, indent=4)`

Reply:
Yeah, that's shorter